### ISSUE TITLE
Error "Product options length does not match variant options length" when adding a product via API

### ISSUE THREAD

**Issue by harshajayaweeraXHJ** [medusajs/medusa#4882]
I'm using <backend url>/admin/products POST endpoint to add products to my store.
This is the request body I'm sending

```
{
    "title": "Sample title",
    "subtitle": "sample sub title",
    "description": "description goes here",
    "status": "draft",
    "thumbnail": "<my image url>",
    "collection_id": "pcol_01GYE........", // This is an existing collection id
    "images": [
        <my image url>
    ],
    "tags": [
        {
            "value": "tag 1"
        }
    ],
    "options": [
        {
            "title": "<option title>"
        }
    ],
    "variants": [
        {
            "title": "<title>",
            "sku": null,
            "inventory_quantity": 999963,
            "prices": [
                {
                    "currency_code": "usd",
                    "amount": 1231
                }
            ],
            "options": [
                {
                    "value": "<option title>"
                }
            ]
        }
    ],
    "sales_channels": [
        {
            "id": "sc_01GY2A........" // this is an existing sales channel
        }
    ]
}
```

When I try this I get following error

```
{
    "type": "invalid_data",
    "message": "Product options length does not match variant options length. Product has 0 and variant has 1.\nCannot read properties of undefined (reading 'length')"
}
```

As you can see in the above request JSON, top level "option" key has a one element and variant object's "option" key also has one element in it's array as well. 

So I'm confuses about the error message. Please clarify

**Comment by olivermrbl** [comment]
What version do you have installed?

I just created a Product from this exact payload using a server running the latest version.

**Comment by harshajayaweeraXHJ** [comment]
My medusa server package.json says following 
```
   "@medusajs/medusa": "^1.10.0",
   "@medusajs/medusa-cli": "^1.3.12",
```

**Comment by harshajayaweeraXHJ** [comment]
@olivermrbl any idea why it is incompatible with my version and what is the correct payload format for my version?

**Comment by olivermrbl** [comment]
I am not sure what is causing your issue. I recommend upgrading your server to run with the latest versions.

We've also shipped tonnes of improvements and features since 1.10.0, which I hope you'll be excited about :)

**Comment by harshajayaweeraXHJ** [comment]
@olivermrbl I would like to but not at the moment as we have a production running application and need to do the upgrade carefully with good testing. But we have to do an urgent integration so we are stuck with this bug.
What is causing this issue. 1.10.0 is a stable release so why this create product endpoint doesn't work

**Comment by harshajayaweeraXHJ** [comment]
I was able to fix this bug by updating my backend to `v1.10.1`. Thanks for the help
